File: api/models/system.go
package models

type System struct {
	Name      string `json:"name"`
	FullName  string `json:"fullName"`
	CoreName  string `json:"coreName"`
	CorePath  string `json:"corePath"`
	Extension string `json:"extension"`
	Cover     string `json:"cover"`
	CoverDark string `json:"coverDark"`

	Games []Game `json:"games"`
}

File: ui/sources/entities/system.js
import { Game } from "./game";

export class System {
	/** @type {string} */
	name;

	/** @type {string} */
	lib_name;

	/** @type {string} */
	core_name;

	/** @type {string} */
	extension;

	/** @type {boolean} */
	standalone;

	/** @type {Game[]} */
	games;
}
